src/panels/lovelace/cards/hui-map-card.ts

```diff
@@ -308,46 +321,50 @@ class HuiMapCard extends LitElement implements LovelaceCard {
     return color;
   }
 
-  private _getEntities = memoizeOne(
-    (
-      states: HassEntities,
-      config: MapCardConfig,
-      configEntities?: MapEntityConfig[]
-    ) => {
-      if (!states || !config) {
-        return undefined;
-      }
+  private _getSourceEntities(
+    states: HassEntities,
+    config?: MapCardConfig
+  ): string[] {
+    if (!states || !config || !config.geo_location_sources) {
+      return [];
+    }
 
-      const geoEntities: string[] = [];
-      if (config.geo_location_sources) {
-        // Calculate visible geo location sources
-        const includesAll = config.geo_location_sources.includes("all");
-        for (const stateObj of Object.values(states)) {
-          if (
-            computeDomain(stateObj.entity_id) === "geo_location" &&
-            (includesAll ||
-              config.geo_location_sources.includes(stateObj.attributes.source))
-          ) {
-            geoEntities.push(stateObj.entity_id);
-          }
-        }
+    const geoEntities: string[] = [];
+    // Calculate visible geo location sources
+    const includesAll = config.geo_location_sources.includes("all");
+    for (const stateObj of Object.values(states)) {
+      if (
+        computeDomain(stateObj.entity_id) === "geo_location" &&
+        (includesAll ||
+          config.geo_location_sources.includes(stateObj.attributes.source))
+      ) {
+        geoEntities.push(stateObj.entity_id);
       }
+    }
+    return geoEntities;
+  }
 
-      return [
-        ...(configEntities || []).map((entityConf) => ({
-          entity_id: entityConf.entity,
-          color: this._getColor(entityConf.entity),
-          label_mode: entityConf.label_mode,
-          focus: entityConf.focus,
-          name: entityConf.name,
-        })),
-        ...geoEntities.map((entity) => ({
+  private _getMapEntities(): HaMapEntity[] {
```

Comment:
I would appreciate a guidance here, as I'm not sure where to put this logic.
The call was added to `shouldUpdate()` instead of `render()`, but it's a side-effect. 
The other option is to implement `attributeChangedCallback` and to update `_mapEntities` when `hass` has changed. But there is no use of this callback in the entire codebase as it should be rarely used...

Reply:
You can use `willUpdate` and check what properties are changed, there you calculate `_mapEntities` and that is used in render.

Reply:
Done